AppCenterAnalytics/AppCenterAnalytics/MSAnalytics.m

```diff
@@ -364,29 +347,35 @@ - (void)resume {
 }
 
 - (void)trackPage:(NSString *)pageName withProperties:(NSDictionary<NSString *, NSString *> *)properties {
-  if (![self isEnabled]) {
-    return;
-  }
+  @synchronized(self) {
+    if (![self canBeUsed] || ![self isEnabled]) {
+      return;
+    }
 
-  // Create an event log.
-  MSPageLog *log = [MSPageLog new];
+    // Create an event log.
+    MSPageLog *log = [MSPageLog new];
 
-  // Set properties of the event log.
-  log.name = pageName;
-  if (properties && properties.count > 0) {
-    log.properties = [self removeInvalidProperties:properties];
-  }
+    // Set properties of the event log.
+    log.name = pageName;
+    if (properties && properties.count > 0) {
+      log.properties = [self removeInvalidProperties:properties];
+    }
 
-  // Send log to log manager.
-  [self sendLog:log flags:MSFlagsDefault];
+    // Send log to log manager.
+    [self sendLog:log flags:MSFlagsDefault];
+  }
 }
 
 - (void)setAutoPageTrackingEnabled:(BOOL)isEnabled {
-  _autoPageTrackingEnabled = isEnabled;
+  @synchronized(self) {
+    _autoPageTrackingEnabled = isEnabled;
```

Comment:
Instead of synchronizing this and `isAutoPageTrackingEnabled`, can you just make `_autoPageTrackingEnabled` `atomic`?